rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -299,11 +299,19 @@ http {
     client_body_buffer_size         {{ $cfg.ClientBodyBufferSize }};
     client_body_timeout             {{ $cfg.ClientBodyTimeout }}s;
 
-    http2_max_field_size            {{ $cfg.HTTP2MaxFieldSize }};
-    http2_max_header_size           {{ $cfg.HTTP2MaxHeaderSize }};
-    http2_max_requests              {{ $cfg.HTTP2MaxRequests }};
     http2_max_concurrent_streams    {{ $cfg.HTTP2MaxConcurrentStreams }};
 
+    {{ if and (gt $http2_max_header_size 0) (gt $http2_max_field_size 0) }}
```

Comment:
The e2e tests fails becuase the variable doesnt exist

https://github.com/kubernetes/ingress-nginx/actions/runs/3758903963/jobs/6387940361#step:12:4329

youll need to check if $cfg.HTTP2MaxHeaderSize and HTTP2MaxFieldSize are > 0

Reply:
like here https://github.com/kubernetes/ingress-nginx/blob/main/rootfs/etc/nginx/template/nginx.tmpl#L649

Reply:
So it should look like this? 

```
{{ if and (gt $cfg.HTTP2MaxHeaderSize 0) (gt $cfg.HTTP2MaxFieldSize 0) }}
```